File: 250216/문자열 놀이/play-with-string.py
import sys

def swap(string, a, b):
    a, b = int(a) - 1, int(b) - 1
    str_list = list(string)  
    str_list[a], str_list[b] = str_list[b], str_list[a]
    return "".join(str_list)

def change(string, a, b):
    return string.replace(a, b)

s, q = sys.stdin.readline().split()
q = int(q)

for _ in range(q):
    cmd, a, b = sys.stdin.readline().split()

    if cmd == "1":
        s = swap(s, a, b)
    else:
        s = change(s, a, b)

    print(s)

File: 250216/문자열 놀이/play-with-string.js
function getNextLine(input) {
    let inputIdx = 0;
    return function () {
        return input[inputIdx++].split(/\s+/);
    };
}

function swap(strArray, a, b) {
    a = Number(a) - 1;
    b = Number(b) - 1;
        let temp = strArray[a];
        strArray[a] = strArray[b];
        strArray[b] = temp;
}

function change(strArray, a, b) {
    for (let i = 0; i < strArray.length; i++) {
        if (strArray[i] === a) strArray[i] = b;
    }
}

const fs = require("fs");

const input = fs.readFileSync(0, "utf8").trim().split("\n");
const nextLine = getNextLine(input);

let [s, q] = nextLine();
q = Number(q);
let strArray = [...s];

const output = [];

while (q--) {
    const [cmd, a, b] = nextLine();
    if (cmd === "1") {
        swap(strArray, a, b);
    } else {
        change(strArray, a, b);
    }
    output.push(strArray.join(""));
}

console.log(output.join("\n"));
